Orange/widgets/visualize/owsieve.py

```diff
@@ -29,11 +29,29 @@ class ChiSqStats:
     pair of attributes. The class is also used for ranking.
     """
     def __init__(self, data, attr1, attr2):
-        self.observed = get_contingency(data, attr1, attr2)
-        self.n = np.sum(self.observed)
-        self.probs_x = self.observed.sum(axis=0) / self.n
-        self.probs_y = self.observed.sum(axis=1) / self.n
-        self.expected = np.outer(self.probs_y, self.probs_x) * self.n
+        attr1 = data.domain[attr1]
+        attr2 = data.domain[attr2]
+        if attr1.is_discrete and not attr1.values or \
+                        attr2.is_discrete and not data.domain[attr2].values:
+            self.observed, self.n = np.nan, len(data)
+            self.probs_x, self.probs_y = [], []
+            if attr1.values:
+                observed_x = np.unique(data.get_column_view(
+                    data.domain.index(attr1))[0], return_counts=True)[1]
+                self.probs_x = observed_x / self.n
+                self.observed = observed_x
+            if attr2.values:
```

Comment:
Would it be better to use `elif` to make it clearer only one of these `if`s can get executed.